src/database/bike.ts

```diff
@@ -79,14 +91,30 @@ export async function getBike(id: number): Promise<Bike> {
 }
 
 export async function patchBike(id: number, req: PatchBikeRequest): Promise<void> {
+    if (_.isEmpty(req)) {
+        throw new Error('user input error');
+    }
+
     const values = [id, req.year, req.make, req.model, req.membershipId];
 
     let result;
     try {
         [result] = await getPool().query<OkPacket>(PATCH_BIKE_SQL, values);
-    } catch (e) {
-        logger.error(`DB error patching bike: ${e}`);
-        throw new Error('internal server error');
+    } catch (e: any) {
+        if ('errno' in e) {
+            switch (e.errno) {
```

Comment:
why does getBike need this error handling? Can a get really violate FKs or a non-null?